doc/man3/ASYNC_start_job.pod

```diff
@@ -250,17 +261,23 @@ The following example demonstrates how to use most of the core async APIs:
                                 pipefds[0], wptr, cleanup);
 
      /*
-      * Normally some external event would cause this to happen at some
+      * Normally some external event (like a network read being ready,
+      * disk access being finished, or some hardware offload operation
+      * completing) would cause this to happen at some
       * later point - but we do it here for demo purposes, i.e.
       * immediately signalling that the job is ready to be woken up after
       * we return to main via ASYNC_pause_job().
       */
      write(pipefds[1], &buf, 1);
 
-     /* Return control back to main */
+     /*
+      * Return control back to main just before calling a blocking
+      * method. The main thread will wait until pipefds[0] is ready
+      * for reading before returning control to this thread.
+      */
      ASYNC_pause_job();
 
-     /* Clear the wake signal */
+     /* Perform the blocking call */
```

Comment:
This change doesn't seem correct. This shouldn't ever block. It just clearing the wake signal that we wrote with the `write` call a few lines above.

Reply:
in this example code, yes, it won't; but then this example code does nothing.
In real world code, this is the blocking call we want to protect against by use of `ASYNC_pause_job()`.